lib/Runtime/Language/SourceTextModuleRecord.cpp

```diff
@@ -175,6 +175,12 @@ namespace Js
             }
 
             OUTPUT_TRACE_DEBUGONLY(Js::ModulePhase, _u("\t>NotifyParentAsNeeded\n"));
+            if (this->promise != nullptr)
+            {
+                SourceTextModuleRecord::ResolveOrRejectDynamicImportPromise(false, this->GetErrorObject(), this->GetScriptContext(), this);
+                this->SetPromise(nullptr);
```

Comment:
>SetPromise [](start = 22, length = 10)

would it be wise to reset the promise to null first and do the ResolveOrReject.. ?